tests/ciscotest.rb

```diff
@@ -62,14 +65,40 @@ def config(*args)
     result
   end
 
+  def ip_address?(ip)
+    return true if IPAddr.new(ip).ipv4?
+  rescue IPAddr::InvalidAddressError
+    false
+  end
+
+  def convert_dns_name(ip)
+    ip_address?(ip) ? ip : Resolv.getaddress(ip)
+  rescue Resolv::ResolvError
+    raise "Unable to resolve name #{ip}. Use static ip to connect instead!"
+  end
+
+  def address_match?(int_ip)
+    # Compare the interface address with the current session address.
+    # and return true if they match.
+    return false if int_ip.nil?
```

Comment:
I agree, but I intentionally left this line because we should return early and not even do the dns conversion, address check etc when nil.  No point.

The last `false` just just a fail safe.